Enable clickable URLs in diagrams

janekska opened the pull request:
Clickable URLs can be defined by adding an option:
"a2s:link":"my/link/to/a/page/or/site"

janekska committed:
Enable clickable URLs in diagrams

Clickable URLs can be defined by adding an option:
"a2s:link":"my/link/to/a/page/or/site"

dhobsd commented on ASCIIToSVG.php:
```diff
@@ -1084,23 +1093,32 @@ public function render() {
     foreach ($this->options as $opt => $val) {
       if (strpos($opt, 'a2s:', 0) === 0) {
-        continue;
+        if ( $opt=='a2s:link' ) {
```
Please fix the `if` style here

dhobsd commented on ASCIIToSVG.php:
```diff
@@ -1084,23 +1093,32 @@ public function render() {
+      $out_p .= "$opt=\"$val\" ";
     }
-    $out .= "d=\"{$path}\" />\n";
+    if ( isset($alnk) ) {
```
Please fix the `if` style here

dhobsd commented on ASCIIToSVG.php:
```diff
@@ -1084,23 +1093,32 @@ public function render() {
       foreach ($this->text as $text) {
-        $text->setID($this->name);
-        $out .= "\t" . $text->render() . "\n";
+        $out_p .= "\t" . $text->render() . "\n";
```
I don't recall what `$text->setID` is for (it's been a while since I did anything with the code), but why is it removed here?

janekska replied:
I guess that is my mistake.
I did wrong assumption that there are the same code in /schplurtz/a2s and I did not catched this line as it was not there as I developed and tested the code on my DokuWiki instalation.
I compared the both sources and found that in your case there is the function setID, so I guess that it is needed here and I put it back.

schplurtz replied:
It is probably correct to remove line 1099. This line re-sets the name property of $text. $text then uses the name as the svg id. Setting the id to the same value for each text object inside a path will result in reused svg id, which is not what is wanted, I guess.
given this input
```
.-------------.
|  abcd       |
|     efgh    |
'-------------'
```
with line 1099 on, both `abcd` and `efgh` get the same id (text0) : 
```svg
<g id="boxes" stroke="black" stroke-width="2" fill="none" >
<g id="group0">
	<path id="path0" filter="url(#dsFilter)" fill="#fff" d="path...." />
	<text x="26.1" y="28.8" id="text0" other-attr-removed-for-clarity>abcd </text>
	<text x="53.1" y="44.8" id="text0" other-attr-removed-for-clarity >efgh </text>
</g>
```

with line 1099 removed, they get different id, that were already correctly assigned by this autoincrement in constructor : `$this->name = self::$id++;`. 
```svg
<g id="boxes" stroke="black" stroke-width="2" fill="none" >
<g id="group0">
	<path id="path0" filter="url(#dsFilter)" fill="#fff" d="path...." />
	<text x="26.1" y="28.8" id="text0" other-attr-removed-for-clarity>abcd </text>
	<text x="53.1" y="44.8" id="text1" other-attr-removed-for-clarity >efgh </text>
</g>
```

dhobsd replied:
Thanks for clarifying

janekska replied:
So, should I remove it?

dhobsd replied:
May as well

dhobsd commented on ASCIIToSVG.php:
```diff
@@ -1084,23 +1093,32 @@ public function render() {
       }
     }
 
+    if ( isset($alnk) ) {
```
Please fix the `if` style here

dhobsd reviewed (changes requested):
Sorry; I forgot about GitHub's review interface. Thanks again for the PR, if you can fix these minor bits, I'll merge it promptly!

janekska committed:
Update ASCIIToSVG.php

Requested fixes from /dhobsd about coding style
and
Suggestion from /schplurtz about W3C URL link definition standarts

janekska committed:
Update ASCIIToSVG.php

Re added missing line.

janekska committed:
Update ASCIIToSVG.php

Removed line with unneeded setID
